features/functests/sctp/sctp.go

```diff
@@ -35,57 +38,143 @@ var _ = Describe("sctp", func() {
 	beforeAll(func() {
 		namespace.Create(testNamespace, clients.K8s)
 		namespace.Clean(testNamespace, clients.K8s)
-		checkForSctpReady(clients.K8s)
-		By("Creating the sctp service")
-		createSctpService(clients.K8s)
 	})
-
-	var _ = Context("Client Server Connection", func() {
+	var _ = Describe("sctp setup", func() {
 		var clientNode string
 		var serverNode string
-		It("Should connect a client pod to a server pod", func() {
-			By("Choosing the nodes for the server and the client")
-			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
-				LabelSelector: "node-role.kubernetes.io/worker=",
-			})
-			Expect(err).ToNot(HaveOccurred())
-			clientNode = nodes.Items[0].ObjectMeta.Labels[hostnameLabel]
-			serverNode = nodes.Items[0].ObjectMeta.Labels[hostnameLabel]
-			if len(nodes.Items) > 1 {
-				serverNode = nodes.Items[1].ObjectMeta.Labels[hostnameLabel]
-			}
+		var activeService *k8sv1.Service
+		var runningPod *k8sv1.Pod
+		var serverPod *k8sv1.Pod
+		var nodes *k8sv1.NodeList
```

Comment:
It would be better to have locally scoped nodes. Here nodes is leaked among tests, but initialized differently each time.
I think it better to keep these variable as more locally scoped as we can.
For `nodes` for example, we can have two different instances, one in `Describe("Test Connectivity Negative", func()` and one other in `var _ = Describe("Test Connectivity", func() { ` scope

This is the same for the other variables. The more local they can be, the better (even if you split them).